CMakeLists.txt

```diff
@@ -42,40 +44,69 @@ if(BUILD_TESTING)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
   endif()
 
-  find_package(ament_cmake_gtest REQUIRED)
+  find_package(ament_cmake_gmock REQUIRED)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 
+  set(extra_test_libraries)
+  set(extra_test_env)
+  set(extra_lib_dirs)
+
+  # This subdirectory extends both extra_test_libraries, extra_test_env, and extra_lib_dirs.
+  add_subdirectory(test/memory_tools)
+
+  set(SKIP_TEST_IF_WIN32 "")
+  if(WIN32)  # (memory tools doesn't do anything on Windows)
+    set(SKIP_TEST_IF_WIN32 "SKIP_TEST")
+  endif()
+
   # Gtests
-  ament_add_gtest(test_split
+  ament_add_gmock(test_allocator test/test_allocator.cpp
+    ENV ${extra_test_env}
+    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+    ${SKIP_TEST_IF_WIN32}
+  )
+  if(TARGET test_allocator)
+    target_link_libraries(test_allocator c_utilities ${extra_test_libraries})
+  endif()
+
+  ament_add_gmock(test_error_handling test/test_error_handling.cpp
+    # Append the directory of libc_utilities so it is found at test time.
+    APPEND_LIBRARY_DIRS "$<TARGET_FILE_DIR:c_utilities>"
+  )
+  if(TARGET test_error_handling)
+    target_link_libraries(test_error_handling ${PROJECT_NAME})
+  endif()
+
+  ament_add_gmock(test_split
     test/test_split.cpp
   )
   if(TARGET test_split)
     target_link_libraries(test_split c_utilities)
   endif()
 
-  ament_add_gtest(test_find
+  ament_add_gmock(test_find
```

Comment:
I thought I should only use gmock or gtest, but not both. I can change them back if that's not the case.